### PR TITLE
[ja] Translate tasks/debug-application-cluster/debug-application into Japanese

### LINKED ISSUE #30873: [ja] Translate tasks/debug-application-cluster/debug-application into Japanese
# This is a Feature Request

## What would you like to be added

Translate [tasks/debug-application-cluster/debug-application](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-application/)  into Japanese.

## Why is this needed

This page has not been translated.

## Comments

/language ja
/assign

### PR DESCRIPTION
**Pull request by ptux**
resolve #30873

### CODE REVIEW

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+<!-- TODO: Now that #11914 is merged, this advice may need to be updated -->
+
+次に確認することは、apiserver上のPodが、作成しようとしたPod（例えば、ローカルマシンのyamlファイル）と一致しているかどうかです。
+例えば、`kubectl get pods/mypod -o yaml > mypod-on-apiserver.yaml` を実行して、元のポッドの説明である`mypod.yaml`とapiserverから戻ってきた`mypod-on-apiserver.yaml`を手動で比較してみてください。
+通常、"apiserver" バージョンには、元のバージョンにはない行がいくつかあります。これは予想されることです。
+しかし、もし元のバージョンにある行がapiserverバージョンにない場合、これはあなたのPod specに問題があることを示している可能性があります。
+
+### レプリケーションコントローラのデバッグ
+
+レプリケーションコントローラはかなり単純なものです。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:89]:
```suggestion
レプリケーションコントローラーはかなり単純なものです。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+彼らはPodを作ることができるか、できないか、どちらかです。
+もしPodを作成できないのであれば、[instructions above](#debugging-pods)を参照して、Podをデバッグしてください。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:91]:
```suggestion
もしPodを作成できないのであれば、[上記の説明](#debugging-pods)を参照して、Podをデバッグしてください。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+また、`kubectl describe rc ${CONTROLLER_NAME}`を使用すると、レプリケーションコントローラに関連するイベントをイントロスペクトすることができます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:92]:
```suggestion
また、`kubectl describe rc ${CONTROLLER_NAME}`を使用すると、レプリケーションコントローラーに関連するイベントを確認することができます。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+### サービスのデバッグ
+
+サービスは、ポッドのセット全体でロードバランシングを提供します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:96]:
```suggestion
Serviceは、Podの集合全体でロードバランシングを提供します。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+しかし、もし元のバージョンにある行がapiserverバージョンにない場合、これはあなたのPod specに問題があることを示している可能性があります。
+
+### レプリケーションコントローラのデバッグ
+
+レプリケーションコントローラはかなり単純なものです。
+彼らはPodを作ることができるか、できないか、どちらかです。
+もしPodを作成できないのであれば、[instructions above](#debugging-pods)を参照して、Podをデバッグしてください。
+また、`kubectl describe rc ${CONTROLLER_NAME}`を使用すると、レプリケーションコントローラに関連するイベントをイントロスペクトすることができます。
+
+### サービスのデバッグ
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:94]:
```suggestion
### Serviceのデバッグ
```

Podに対してServiceですね

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+サービスは、ポッドのセット全体でロードバランシングを提供します。
+サービスが正しく動作しない原因には、いくつかの一般的な問題があります。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:97]:
```suggestion
Serviceが正しく動作しない原因には、いくつかの一般的な問題があります。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+以下の手順は、サービスの問題をデバッグするのに役立つはずです。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:99]:
```suggestion
以下の手順は、Serviceの問題をデバッグするのに役立つはずです。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+まず、サービスのエンドポイントが存在することを確認します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:101]:
```suggestion
まず、Serviceに対応するEndpointが存在することを確認します。
```
この場合Endpointリソースを指していると思います

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+全てのサービスオブジェクトに対して、apiserverは `endpoints` リソースを利用できるようにします。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:102]:
```suggestion
全てのServiceオブジェクトに対して、apiserverは `endpoints` リソースを利用できるようにします。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+このリソースは次のようにして見ることができます。
+
+```shell
+kubectl get endpoints ${SERVICE_NAME}
+```
+
+エンドポイントがサービスのメンバーとして想定されるPod数と一致していることを確認してください。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:109]:
```suggestion
EndpointがServiceのメンバーとして想定されるPod数と一致していることを確認してください。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+例えば、3つのレプリカを持つnginxコンテナ用のサービスであれば、サービスのエンドポイントには3つの異なるIPアドレスが表示されるはずです。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:110]:
```suggestion
例えば、3つのレプリカを持つnginxコンテナ用のServiceであれば、ServiceのEndpointには3つの異なるIPアドレスが表示されるはずです。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+
+#### サービスにエンドポイントがありません
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:112]:
```suggestion
#### Serviceに対応するEndpointがありません
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+
+エンドポイントが見つからない場合は、サービスが使用しているラベルを使用してポッドをリストアップしてみてください。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:114]:
```suggestion
Endpointが見つからない場合は、Serviceが使用しているラベルを使用してPodをリストアップしてみてください。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+
+```yaml
+...
+spec:
+  - selector:
+     name: nginx
+     type: frontend
+```
+
+セレクタに一致するポッドを一覧表示するには、次のコマンドを使用します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:125]:
```suggestion
セレクタに一致するPodを一覧表示するには、次のコマンドを使用します。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+
+```shell
+kubectl get pods --selector=name=nginx,type=frontend
+```
+
+リストがサービスを提供する予定のPodと一致することを確認します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:131]:
```suggestion
リストがServiceを提供する予定のPodと一致することを確認します。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+Podの`containerPort`がサービスの`targetPort`と一致することを確認します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:132]:
```suggestion
Podの`containerPort`がServiceの`targetPort`と一致することを確認します。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+
+#### ネットワークトラフィックが転送されません
+
+詳しくは[debugging service](/docs/tasks/debug-application-cluster/debug-service/)を参照してください。
+
+## {{% heading "whatsnext" %}}
+
+上記のいずれの方法でも問題が解決しない場合は、以下の手順に従ってください。
+[Debugging Service document](/docs/tasks/debug-application-cluster/debug-service/)で、`Service` が実行されていること、`Endpoints`があること、`Pods`が実際にサービスを提供していること、DNS が機能していること、IPtables ルールがインストールされていること、kube-proxyが誤作動を起こしていないようなことを確認してください。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:141]:
```suggestion
[Debugging Service document](/docs/tasks/debug-application-cluster/debug-service/)で、`Service` が実行されていること、`Endpoints`があること、`Pods`が実際にサービスを提供していること、DNS が機能していること、IPtablesルールがインストールされていること、kube-proxyが誤作動を起こしていないようなことを確認してください。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+
+トラブルシューティングドキュメント](/docs/tasks/debug-application-cluster/troubleshooting/)に詳細が記載されています。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:143]:
```suggestion
[トラブルシューティングドキュメント](/docs/tasks/debug-application-cluster/troubleshooting/)に詳細が記載されています。
```

**Review by nasa9084 — CHANGES_REQUESTED** [review]

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 19]
```diff
@@ -0,0 +1,144 @@
...
+kubectl describe pods ${POD_NAME}
+```
+
+Pod内のコンテナの状態を見てください。
+すべて`Running`ですか？ 最近、再起動がありましたか？
+ポッドの状態に応じてデバッグを続けます。
+
+#### PodがPendingのまま
+
+Podが`Pending` で止まっている場合、それはノードにスケジュールできないことを意味します。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:40]:
```suggestion
Podが`Pending`で止まっている場合、それはノードにスケジュールできないことを意味します。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 20]
```diff
@@ -0,0 +1,144 @@
...
+一般に、これはある種のリソースが不十分で、スケジューリングできないことが原因です。
+上の`kubectl describe ...`コマンドの出力を見てください。
+
+なぜあなたのPodをスケジュールできないのか、スケジューラからのメッセージがあるはずです。
+理由は以下の通りです。
+
+* **リソースが不足しています。** クラスタのCPUまたはメモリーを使い果たした可能性があります。Podを削除するか、リソース要求を調整するか、クラスタに新しいノードを追加する必要があります。詳しくは[Compute Resources document](/docs/concepts/configuration/manage-resources-containers/)を参照してください。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:47]:
```suggestion
* **リソースが不足しています。** クラスターのCPUまたはメモリーを使い果たしている可能性があります。Podを削除するか、リソースの要求値を調整するか、クラスターに新しいノードを追加する必要があります。詳しくは[Compute Resources document](/ja/docs/concepts/configuration/manage-resources-containers/)を参照してください。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 21]
```diff
@@ -0,0 +1,144 @@
...
+
+* **あなたが使用しているのは`hostPort`**です。Podを`hostPort`にバインドすると、そのPodがスケジュールできる場所が限定されます。ほとんどの場合、`hostPort`は不要なので、Serviceオブジェクトを使ってPodを公開するようにしてください。もし`hostPort` が必要な場合は、Kubernetes クラスタのノード数だけ Pod をスケジュールすることができます。
+
+
+#### Podがwaitingのまま
+
+Podが `Waiting` 状態で止まっている場合、ワーカーノードにスケジュールされていますが、そのマシン上で実行することはできません。この場合も、`kubectl describe ...` の情報が参考になるはずです。`Waiting`状態のポッドの最も一般的な原因は、イメージの取り出しに失敗することです。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:54]:
```suggestion
Podが`Waiting`状態で止まっている場合、ワーカーノードにスケジュールされていますが、そのノード上で実行することができません。この場合も、`kubectl describe ...`の情報が参考になるはずです。`Waiting`状態のPodの最も一般的な原因は、コンテナイメージのプルに失敗することです。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 22]
```diff
@@ -0,0 +1,144 @@
...
+
+確認すべきことは3つあります。
+
+* イメージの名前が正しいかどうか確認してください。
+* イメージをリポジトリにプッシュしましたか？
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:59]: Isn't this the registry?
However, the upstream documentation describes it as a repository.
"Repository" is a collection of container images or other artifacts in the registry that have the same name but different tags.

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 23]
```diff
@@ -0,0 +1,144 @@
...
+```shell
+kubectl get pods --selector=name=nginx,type=frontend
+```
+
+リストがサービスを提供する予定のPodと一致することを確認します。
+Podの`containerPort`がサービスの`targetPort`と一致することを確認します。
+
+#### ネットワークトラフィックが転送されません
+
+詳しくは[debugging service](/docs/tasks/debug-application-cluster/debug-service/)を参照してください。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:136]:
```suggestion
詳しくは[Serviceのデバッグ](/ja/docs/tasks/debug-application-cluster/debug-service/)を参照してください。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md (hunk 17 above)
Comment (riita10069) [reply content/ja/docs/tasks/debug-application-cluster/debug-application.md:141]:
```suggestion
[Serviceのデバッグに関するドキュメント](/ja/docs/tasks/debug-application-cluster/debug-service/)で、`Service` が実行されていること、`Endpoints`があること、`Pods`が実際にサービスを提供していること、DNS が機能していること、IPtablesルールがインストールされていること、kube-proxyが誤作動を起こしていないようなことを確認してください。
```

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md (hunk 22 above)
Comment (ptux) [reply content/ja/docs/tasks/debug-application-cluster/debug-application.md:57]: I think you are right.
And I have created PR for the upstream.

> https://github.com/kubernetes/website/pull/31170

Let's wait until the PR merged.

Comment (ptux) [reply content/ja/docs/tasks/debug-application-cluster/debug-application.md]: Got #31170 PR merged.
And changed `リポジトリ` to `レジストリ`.

File: content/ja/docs/tasks/debug-application-cluster/debug-application.md [hunk 24]
```diff
@@ -0,0 +1,142 @@
...
+このガイドは、Kubernetesにデプロイされ、正しく動作しないアプリケーションをユーザーがデバッグするためのものです。
+これは、自分のクラスターをデバッグしたい人のためのガイドでは *ありません*。
+そのためには、[debug-cluster](/docs/tasks/debug-application-cluster/debug-cluster)を確認する必要があります。
+
+<!-- body -->
+
+## 問題の診断
+
+トラブルシューティングの最初のステップは切り分けです。何が問題なのでしょうか？
+Podなのか、レプリケーションコントローラなのか、それともサービスなのか？
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/debug-application.md:17]:
```suggestion
Podなのか、レプリケーションコントローラーなのか、それともサービスなのか？
```

**Review by nasa9084 — CHANGES_REQUESTED** [review]

**Comment by ptux** [comment]
@nasa9084 @inductor 

You'd better merge the PR with squash.

`/label tide/merge-method-squash`

**Pull request merged**